basescript/basescript.py

```diff
@@ -31,17 +32,26 @@ def __init__(self):
         self.args = self.parser.parse_args()
 
         self.hostname = socket.gethostname()
-        self.log = self.init_logger(self.args.log_level)
+        self.log = self.init_logger()
 
         args = { n: getattr(self.args, n) for n in vars(self.args) }
+        args['func'] = self.args.func.func_name
         self.log.debug("basescript init", **args)
 
     def start(self):
         '''
         Starts execution of the script
         '''
         # invoke the appropriate sub-command as requested from command-line
-        self.args.func()
+        try:
+            self.args.func()
+        except:
```

Comment:
@algrebe a `KeyboardInterrupt` represents an intentional termination of a script. Should we not exit with code 0 for this? Also, What if the code does sys.exit(0) somewhere (Representing successful termination). Is it right to return error code 1 for that?